### ISSUE TITLE
'Block all unencrypted requests' shows as enabled on browser startup and is ticked, but is really disabled and has to be re-ticked. 

### ISSUE THREAD

**Issue by taoje** [EFForg/https-everywhere#13257]
Type: code issue

When 'block all unencrypted requests' was enabled in the old HTTPS Everywhere, on the browser restarting it would remain enabled by default. However in the newest version HTTPS Everywhere _shows_ 'block all unencrypted requests' is enabled but really it isn't: HTTP websites load successfully. To enable, the tick box must be un-ticked and then re-ticked. 

In Firefox downloading HTTPS Everywhere from the official website alongside the not working one in browser helped the issue. However Chrome doesn't allow two plugins for the same browser and has - since today - 21st October 2017 - started to do the same thing: show 'Block all unencrypted requests' as enabled, the tick box ticked, while it is not. 

This issue has been long standing for months now and I expected others would report and it would be fixed. Also, the reporting process on Github put me off. But since I have not seen this happen, I am reporting it now. Thank you.

**Comment by jeremyn** [comment]
I can't reproduce this problem with Firefox 56.0 and HTTPS Everywhere 2017.10.4. I'm using the EFF test site http://freerangekitten.com which supports both HTTPS and plain HTTP, and when I restart Firefox with `Block all unencrypted requests` enabled and the `Freerangekitten.com` rule disabled, the request is correctly rewritten to HTTPS. Can you reproduce this problem with http://freerangekitten.com ?

There are some [remaining problems](https://github.com/EFForg/https-everywhere/issues?q=is%3Aopen+is%3Aissue+label%3Awebextensions-firefox) with the WebExtensions version of the add-on, including issue https://github.com/EFForg/https-everywhere/issues/12730 where HTTPS Everywhere doesn't remember settings if you clear all your Firefox data (including private browsing mode). So please make sure you aren't using private browsing mode when reproducing this problem.

**Comment by koops76** [comment]
Don't tell me this is happening again.

**Comment by taoje** [comment]
@jeremyn, it only happened after shutting down and then turning the computer back on again and initially starting Firefox or Chrome. I had browser startup pages set, one of which was http-only and they would load: often on Firefox and only yesterday on Chrome. 

When I had both the old and new HTTPS Everywhere icon next to each other in Firefox, the old one (that loaded the http-only page) would turn a _light blue_ compared to the blue of the icon now - before turning red to say 'block all unencrypted connections' was on but it wasn't; _both_ for the initial startup pages and for subsequent http-only pages that I opened afterwards to check.  

I do clear some of my browsing data but I don't use private mode. I think you are right, that may be the cause. 

@koops76, I've started up both Firefox and Chrome after booting on and they seem to be working okay. It was only that one time yesterday Chrome didn't block http-only websites. 

@jeremyn and @koops76, thank you for all the help.
